Objective-C/CBLArray.mm

```diff
@@ -99,6 +100,15 @@ - (void) fl_encodeToFLEncoder: (FLEncoder)enc {
     abort();
 }
 
+#pragma mark - toJSON
+
+- (NSString*) toJSON {
+    JSONEncoder enc;
+    _array.encodeTo(enc);
+    auto data = enc.finish();
+    return slice2string(data);
```

Comment:
In CBLDictionary's toJSON, the code is under _sharedLock. Can you check if the lock is needed?